### PR TITLE
Issue 0093 add lint at the actions

### PR DESCRIPTION
**Pull request by Olender**
### Description
This PR addresses **Issue #93** by adding a linting step to the GitHub Actions workflow to ensure code quality and maintain consistency.

### Changes Made
1. **GitHub Actions Configuration (`lint.yml`):**
   - Added a new workflow named "Run lint."
   - Configured the workflow to trigger on:
     - Push events to the `master` branch.
     - Pull requests.

2. **Makefile Adjustments:**
   - Added a `format` target to format the codebase using `autopep8` with `setup.cfg` as the configuration file.
   - Added a `lint` target to run `flake8` on `setup.py`, `spyro/`, and `test/*.py`.

3. **Codebase Formatting:**
   - Various files have been formatted according to the linting rules defined.

### Additional Notes
- Developers should run `make lint` locally to catch issues before pushing to reduce CI errors.
- The formatting step (`make format`) ensures code consistency based on the settings defined in `setup.cfg`.

### CODE REVIEW

**Comment by Olender** [comment]
I based the flake8 parameters and the lint.yml file configuration on the ones used by the Firedrake project (see [Firedrake's setup.cfg](https://github.com/firedrakeproject/firedrake/blob/master/setup.cfg) for reference).

File: spyro/receivers/changing_coordinates.py [hunk 1]
```diff
@@ -763,4 +763,4 @@ def change_to_reference_hexa(p, cell_vertices):
     # pny = px * a21 + py * a22 + pz * a23 + a24
     # pnz = px * a31 + py * a32 + pz * a33 + a34
 
-    return (pnx, pny, pnz)
+    # return (pnx, pny, pnz)
```
Comment (SouzaEM) [inline spyro/receivers/changing_coordinates.py:766]: Here I think that removing the big block of comments is more appropriate to make the code easier to navigate.

**Review by SouzaEM — APPROVED** [review]
Great work setting up the linter in the CI system! My only recommendation is to avoid big blocks of comments in the future. As the code is under version control, it is always possible to check previous implementations without leaving them commented.

File: spyro/receivers/changing_coordinates.py (hunk 1 above)
Comment (Olender) [reply spyro/receivers/changing_coordinates.py]: I have temporarily removed this section in the latest commit. However, I plan on adding an improved version if we ever work with hexahedrals not based on extruded meshes. Thanks for the suggestion!

**Review by SouzaEM — APPROVED** [review]

**Pull request merged**